src/solver/simulation/remix-storage/storage-for-remix.cpp

```diff
@@ -0,0 +1,238 @@
+                                                       inflows,
+                                                       overflows,
+                                                       injection,
+                                                       lowRuleCurve,
+                                                       upRuleCurve,
+                                                       initLevel,
+                                                       withdrawalEff,
+                                                       injectionEff);
+}
+
+StorageForRemixNoLevels::StorageForRemixNoLevels(std::vector<double>& withdrawal,
+                                                 std::vector<double>& unsupE,
+                                                 const std::vector<double> Pmax,
+                                                 const std::vector<double> Pmin):
+    withdrawal_(withdrawal),
+    initWithdrawal_(withdrawal),
+    unsupE_(unsupE),
+    pmax_(Pmax),
+    pmin_(Pmin)
+{
+    checkInput(unsupE_.size());
+}
+
+double StorageForRemixNoLevels::maxExchange(unsigned hourOfMaxGen, unsigned hourOfMinGen) const
+{
+    // Max amont we can take from hydro withdrawal, at hour when the total
+    // production reaches a max.
+    double boundAtMax = withdrawal_[hourOfMaxGen] - pmin_[hourOfMaxGen];
+    // Max amont we can add to hydro withdrawal, at hour when the total
+    // production reaches a min.
+    double boundAtMin = std::min(
+      {pmax_[hourOfMinGen] - withdrawal_[hourOfMinGen], unsupE_[hourOfMinGen]});
+
+    return std::min(boundAtMax, boundAtMin);
+}
+
+void StorageForRemixNoLevels::checkInput(size_t size)
+{
+    std::vector<size_t> sizes = {size, withdrawal_.size(), pmin_.size(), pmax_.size()};
+
+    if (!std::ranges::all_of(sizes, [&sizes](const size_t s) { return s == sizes.front(); }))
+    {
+        throw std::invalid_argument(error_msg_start + "arrays of different sizes");
+    }
+
+    if (!(withdrawal_ <= pmax_ + TOLERANCE))
+    {
+        throw std::invalid_argument(error_msg_start
+                                    + "Storage withdrawal not smaller than Pmax everywhere");
+    }
+
+    if (!(pmin_ - TOLERANCE <= withdrawal_))
+    {
+        throw std::invalid_argument(error_msg_start
+                                    + "Storage withdrawal not greater than Pmin everywhere");
+    }
+}
+
+void StorageForRemixNoLevels::update()
+{
+}
+
+const std::vector<double>& StorageForRemixNoLevels::initWithdrawal()
+{
+    return initWithdrawal_;
+}
+
+std::vector<double>& StorageForRemixNoLevels::withdrawal()
+{
+    return withdrawal_;
+}
+
+StorageForRemixWithLevels::StorageForRemixWithLevels(std::vector<double>& withdrawal,
+                                                     std::vector<double>& unsupE,
+                                                     std::vector<double>& levels,
+                                                     const std::vector<double> Pmax,
```

Comment:
if you take the exemple of **Pmax**, it was just copied when calling the current constructor.
Then we forward it to constructor of **StorageForRemixNoLevels** by a copy, and then we store it by making another copy.